src/Http/Cookie/Cookie.php

```diff
@@ -0,0 +1,468 @@
+        }
+    }
+
+    /**
+     * Gets the raw cookie value
+     *
+     * @return string
+     */
+    public function getValue()
+    {
+        return $this->value;
+    }
+
+    /**
+     * Sets the raw cookie data
+     *
+     * @param string|array $value Value of the cookie to set
+     * @return $this
+     */
+    public function setValue($value)
+    {
+        if (is_array($value)) {
+            $this->isExpanded = true;
+        }
+
+        $this->value = $value;
+
+        return $this;
+    }
+
+    /**
+     * Sets the path
+     *
+     * @param string|null $path Sets the path
+     * @return $this
+     */
+    public function setPath($path)
+    {
+        $this->path = $path;
+
+        return $this;
+    }
+
+    /**
+     * Sets the domain
+     *
+     * @param string $domain Domain to set
+     * @return $this
+     */
+    public function setDomain($domain)
+    {
+        $this->domain = $domain;
+
+        return $this;
+    }
+
+    /**
+     * Check if the cookie is secure
+     *
+     * @return bool
+     */
+    public function isSecure()
+    {
+        return $this->secure;
+    }
+
+    /**
+     * Check if the cookie is HTTP only
+     *
+     * @return bool
+     */
+    public function isHttpOnly()
+    {
+        return $this->isHttpOnly();
+    }
+
+    /**
+     * Sets the expiration date
+     *
+     * @param \DateTimeInterface $dateTime Date time object
+     * @return $this
+     */
+    public function expiresAt(DateTimeInterface $dateTime)
+    {
+        $this->expiresAt = (int)$dateTime->format('U');
+
+        return $this;
+    }
+
+    /**
+     * Checks if a value exists in the cookie data
+     *
+     * @param string $path Path to check
+     * @return bool
+     */
+    public function check($path)
+    {
+        return Hash::check($this->value, $path);
+    }
+
+    /**
+     * Writes data to the cookie
+     *
+     * @param string $path Path to write to
+     * @param mixer $value Value to write
+     * @return $this
+     */
+    public function write($path, $value)
+    {
+        if (!$this->isExpanded) {
+            throw new RuntimeException('The Cookie data has not been expanded');
+        }
+
+        Hash::insert($this->value, $path, $value);
+
+        return $this;
+    }
+
+    /**
+     * Read data from the cookie
+     *
+     * @param string $path Path to read the data from
+     * @return mixed
+     */
+    public function read($path = null)
+    {
+        if (!$this->isExpanded) {
+            throw new RuntimeException('The Cookie data has not been expanded');
+        }
+
+        if ($path === null) {
+            return $this->value;
+        }
+
+        return Hash::get($this->value, $path);
+    }
+
+    /**
+     * Sets the cookies date to a far future so it will virtually never expire
+     *
+     * @return $this
+     */
+    public function willNeverExpire()
+    {
+        $this->expiresAt = Chronos::now()->addYears(50)->format('U');
```

Comment:
Maybe set this to maximum 32 bit timestamp because of [this](https://en.wikipedia.org/wiki/Year_2038_problem)?